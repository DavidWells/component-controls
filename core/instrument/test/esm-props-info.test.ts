import path from 'path';
import { loadStoriesTests } from './loadTestFiles';

describe('esm-props-info', () => {
  loadStoriesTests(
    {
      stories: { sourceFiles: false },
      propsLoaders: [
        { name: '@component-controls/react-docgen-info', test: /\.(js|jsx)$/ },
        {
          name: '@component-controls/react-docgen-typescript-info',
          test: /\.(ts|tsx)$/,
        },
        ,
      ],
      components: {
        sourceFiles: false,
        resolveFile: (componentName, filePath) => {
          if (filePath.includes('@component-controls/blocks')) {
            const resolved = path.resolve(
              path.dirname(filePath),
              `../../../../ui/blocks/src/${componentName}/${componentName}.tsx`,
            );
            return resolved;
          }
          return filePath;
        },
      },
    },
    ['esm', 'props-info'],
  );
});
