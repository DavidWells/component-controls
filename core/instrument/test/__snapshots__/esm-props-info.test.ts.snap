// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`esm-props-info resolve-node-modules.jsx 1`] = `
Object {
  "components": Object {
    "26090286b0d42fa7a92590e6843f1687": Object {
      "from": "@component-controls/blocks",
      "importedName": "PropsTable",
      "imports": Object {
        "@component-controls/components": Array [
          Object {
            "importedName": "TableProps",
            "name": "TableProps",
          },
        ],
        "@component-controls/store": Array [
          Object {
            "importedName": "StoryContextProvider",
            "name": "StoryContextProvider",
          },
          Object {
            "importedName": "ControlsContextStoryProvider",
            "name": "ControlsContextStoryProvider",
          },
        ],
        "react": Array [
          Object {
            "importedName": "FC",
            "name": "FC",
          },
        ],
        "react-table": Array [
          Object {
            "importedName": "Column",
            "name": "Column",
          },
        ],
        "theme-ui": Array [
          Object {
            "importedName": "jsx",
            "name": "jsx",
          },
        ],
      },
      "name": "PropsTable",
      "package": "5633741e98073052cbece6fcd32011e3",
      "request": "/Users/atanasster/component-controls/ui/blocks/src/PropsTable/PropsTable.tsx",
    },
  },
  "doc": Object {
    "component": "PropsTable",
    "components": Object {},
    "componentsLookup": Object {
      "PropsTable": "26090286b0d42fa7a92590e6843f1687",
    },
    "date": 2020-10-21T01:23:26.430Z,
    "dateModified": 2020-10-21T01:23:26.430Z,
    "package": "942f7c3f19d9e197f86041eed5e46838",
    "title": "Storybook/Blocks/PropsTable",
  },
  "packages": Object {
    "5633741e98073052cbece6fcd32011e3": Object {
      "dependencies": Object {
        "@component-controls/components": "^1.39.1",
        "@component-controls/core": "^1.39.1",
        "@component-controls/editors": "^1.39.1",
        "@component-controls/render": "^1.39.1",
        "@component-controls/store": "^1.39.1",
        "@theme-ui/css": "^0.4.0-rc.1",
        "copy-to-clipboard": "^3.2.1",
        "fuse.js": "^6.4.1",
        "global": "^4.3.2",
        "js-string-escape": "^1.0.1",
        "query-string": "^6.13.5",
        "react-frame-component": "^4.1.2",
        "react-resize-detector": "^5.0.6",
        "recoil": "^0.0.10",
        "stringify-object": "^3.3.0",
        "theme-ui": "^0.4.0-rc.1",
      },
      "devDependencies": Object {
        "@component-controls/jest-snapshots": "^1.39.1",
        "@component-controls/ts-markdown-docs": "^1.37.0",
        "@primer/octicons-react": "^10.0.0",
        "@theme-ui/presets": "^0.3.0",
        "@types/mdx-js__react": "^1.5.1",
        "@types/query-string": "^6.3.0",
        "@types/react": "^16.9.34",
        "@types/react-frame-component": "^4.1.1",
        "@types/react-resize-detector": "^4.2.0",
        "@types/recoil": "^0.0.1",
        "@types/stringify-object": "^3.2.0",
        "cross-env": "^5.2.1",
        "eslint": "^6.5.1",
        "react": "^16.13.1",
        "react-dom": "^16.13.1",
        "react-table": "^7.0.0",
        "typescript": "^4.0.5",
      },
      "fileHash": "5633741e98073052cbece6fcd32011e3",
      "name": "@component-controls/blocks",
      "peerDependencies": Object {
        "@primer/octicons-react": ">= 10.0.0",
        "react": "^16.8.0 || ^17",
        "react-dom": "^16.8.0 || ^17",
        "react-table": ">= 7.0.0",
      },
      "repository": Object {
        "browse": "https://github.com/ccontrols/component-controls/tree/master/ui/blocks/dist/index.js",
        "docs": "https://github.com/ccontrols/component-controls/tree/master#readme",
        "issues": "https://github.com/ccontrols/component-controls/issues",
      },
      "version": "1.39.1",
    },
    "942f7c3f19d9e197f86041eed5e46838": Object {
      "dependencies": Object {
        "@babel/generator": "^7.9.4",
        "@babel/parser": "^7.9.4",
        "@babel/traverse": "^7.9.0",
        "@component-controls/core": "^1.39.1",
        "@hutson/parse-repository-url": "^5.0.0",
        "@mdx-js/loader": "^1.5.5",
        "@mdx-js/react": "^1.6.5",
        "camelcase": "^6.0.0",
        "deepmerge": "^4.2.2",
        "find-cache-dir": "^3.3.1",
        "gray-matter": "^4.0.2",
        "hosted-git-info": "^3.0.4",
        "js-string-escape": "^1.0.1",
        "prettier": "^1.19.1",
        "read-package-json": "^2.1.1",
        "remark-emoji": "^2.1.0",
        "remark-images": "^2.0.0",
        "resolve": "^1.15.1",
        "typescript": "^4.0.5",
      },
      "devDependencies": Object {
        "@babel/types": "^7.9.0",
        "@component-controls/ts-markdown-docs": "^1.37.0",
        "@rollup/plugin-node-resolve": "^7.1.1",
        "@types/find-cache-dir": "^3.2.0",
        "@types/hosted-git-info": "^3.0.0",
        "@types/jest": "^26.0.10",
        "cross-env": "^5.2.1",
        "eslint": "^6.5.1",
        "jest": "^26.4.2",
      },
      "fileHash": "942f7c3f19d9e197f86041eed5e46838",
      "name": "@component-controls/instrument",
      "peerDependencies": undefined,
      "repository": Object {
        "browse": "https://github.com/ccontrols/component-controls/tree/master/core/instrument/test/fixtures/esm/props-info/resolve-node-modules.jsx",
        "docs": "https://github.com/ccontrols/component-controls/tree/master#readme",
        "issues": "https://github.com/ccontrols/component-controls/issues",
      },
      "version": "1.39.1",
    },
  },
  "stories": Object {},
  "transformed": "import { PropsTable } from '@component-controls/blocks';

export default {
  title: 'Storybook/Blocks/PropsTable',
  component: PropsTable,
};
",
}
`;
